mllib/src/test/scala/org/apache/spark/mllib/tree/DecisionTreeSuite.scala

```diff
@@ -331,12 +336,62 @@ class DecisionTreeSuite extends SparkFunSuite with MLlibTestSparkContext {
     assert(topNode.impurity !== -1.0)
 
     // set impurity and predict for child nodes
-    assert(topNode.leftNode.get.predict.predict === 0.0)
-    assert(topNode.rightNode.get.predict.predict === 1.0)
+    if (topNode.leftNode.get.predict.predict === 0.0) {
+      assert(topNode.rightNode.get.predict.predict === 1.0)
+    } else {
+      assert(topNode.leftNode.get.predict.predict === 1.0)
+      assert(topNode.rightNode.get.predict.predict === 0.0)
+    }
     assert(topNode.leftNode.get.impurity === 0.0)
     assert(topNode.rightNode.get.impurity === 0.0)
   }
 
+  test("Use soft prediction for binary classification with ordered categorical features") {
```

Comment:
Hmm, I just want a test case to show it actually order the bins by soft prediction. Although @jkbradley suggested we should use directly `binsToBestSplit`, but in order to do that, we also need to expose many details of `findBestSplits` too, e.g., `binSeqOp`, `getNodeToFeatures` and `partitionAggregates`...etc.